crates/containerd-shim-wasm/src/wasm_libcontainer/mod.rs

```diff
@@ -0,0 +1,80 @@
+//! Common utilities for the containerd shims, including wasmtime and wasmedge shim.
+
+use crate::sandbox::error::Error;
+use anyhow::{bail, Context, Result};
+use libcontainer::container::Container;
+use std::{
+    fs::{self, OpenOptions},
+    io::ErrorKind,
+    os::fd::{IntoRawFd, RawFd},
+    path::{Path, PathBuf},
+};
+
+/// Loads the container state from the given root path.
+pub fn load_container<P: AsRef<Path>>(root_path: P, container_id: &str) -> Result<Container> {
+    let container_root = construct_container_root(root_path, container_id)?;
+    if !container_root.exists() {
+        bail!("container {} does not exist.", container_id)
+    }
+
+    Container::load(container_root)
```

Comment:
Following up on the conversation around modules and crates, now that the source is back.  

This is the only line that feels like it is specific to libcontainer.  The other functions are useful regardless (I needed to canonicalize the root path in windows too).  

I like pulling those other functions into this crate but not sure they need to be behind a flag.  And since it is one line to load the root maybe don't need duplicate this function?

Might be missing something, just some thoughts

Reply:
Yeah maybe I can get rid of this entirely. Let me do some refactoring work there

Reply:
This commit https://github.com/containerd/runwasi/pull/142/commits/b606b25373ddd3b933099064dd49fc18fe280359 removes the feature flag entirely.

Reply:
I still feel that, although now these functions don't depend on `libcontainer`, the assumption that there is a `root_path` to a container runtime's state is pretty libcontainer dependent, no?

Reply:
From what I can tell the only part that is libcontainer dependent is adding the `container_id` to the end of the cannocal path, since ` Container::load(container_root)` and adding `cointainer_id` are both one line, I don't know that sharing logic in a common script is worth it, and I think it would be better to just inline those in the respective shims.

Reply:
Do you think that `get_instance_root`, `instance_exists` and `maybe_open_stdio` are worth in a common script?